src/firewall/server/dispatcher.py

```diff
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+import os
+import subprocess
+
+from firewall.core.logger import log
+from firewall.config import FIREWALLD_DISPATCHER, ETC_FIREWALLD_DISPATCHER, COMMANDS
+
+
+def run_dispatcher(interface, signal, args):
+    def _dispatchers(directories):
+        for directory in directories:
+            if not os.path.isdir(directory):
+                continue
+            for file in os.listdir(directory):
```

Comment:
there is a race, that `directory` just got deleted after you check for it. Or maybe you can even get an exception when the permissions are lacking.

`_dispatchers()` shouldn't raise an exception.

Wrap a try/except around listdir().